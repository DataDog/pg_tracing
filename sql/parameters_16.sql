-- Saturate the parameter buffer with extended protocol
SET pg_tracing.max_parameter_size=1;
/*traceparent='00-00000000000000000000000000000001-0000000000000001-01'*/ select $1, $2, $3 \bind 1 2 3 \g
SELECT span_operation, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000001';

SET pg_tracing.max_parameter_size=2;
/*traceparent='00-00000000000000000000000000000002-0000000000000002-01'*/ select $1, $2, $3 \bind 1 2 3 \g
SELECT span_operation, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000002';

SET pg_tracing.max_parameter_size=3;
/*traceparent='00-00000000000000000000000000000003-0000000000000003-01'*/ select $1, $2, $3 \bind 1 2 3 \g
SELECT span_operation, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000003';

-- Cleanup
CALL clean_spans();
CALL reset_settings();
