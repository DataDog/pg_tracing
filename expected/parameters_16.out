-- Saturate the parameter buffer with extended protocol
SET pg_tracing.max_parameter_size=1;
/*traceparent='00-00000000000000000000000000000001-0000000000000001-01'*/ select $1, $2, $3 \bind 1 2 3 \g
 ?column? | ?column? | ?column? 
----------+----------+----------
 1        | 2        | 3
(1 row)

SELECT span_operation, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000001';
  span_operation   | parameters  | lvl 
-------------------+-------------+-----
 select $1, $2, $3 | {1,...,...} |   1
 Planner           |             |   2
 ExecutorRun       |             |   2
 Result            |             |   3
(4 rows)

SET pg_tracing.max_parameter_size=2;
/*traceparent='00-00000000000000000000000000000002-0000000000000002-01'*/ select $1, $2, $3 \bind 1 2 3 \g
 ?column? | ?column? | ?column? 
----------+----------+----------
 1        | 2        | 3
(1 row)

SELECT span_operation, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000002';
  span_operation   | parameters  | lvl 
-------------------+-------------+-----
 select $1, $2, $3 | {1,...,...} |   1
 Planner           |             |   2
 ExecutorRun       |             |   2
 Result            |             |   3
(4 rows)

SET pg_tracing.max_parameter_size=3;
/*traceparent='00-00000000000000000000000000000003-0000000000000003-01'*/ select $1, $2, $3 \bind 1 2 3 \g
 ?column? | ?column? | ?column? 
----------+----------+----------
 1        | 2        | 3
(1 row)

SELECT span_operation, parameters, lvl from peek_ordered_spans where trace_id='00000000000000000000000000000003';
  span_operation   | parameters | lvl 
-------------------+------------+-----
 select $1, $2, $3 | {1,2,...}  |   1
 Planner           |            |   2
 ExecutorRun       |            |   2
 Result            |            |   3
(4 rows)

-- Cleanup
CALL clean_spans();
CALL reset_settings();
